proposals/getposition-sidepanel.md

```diff
@@ -0,0 +1,130 @@
+# Add API to Query Side Panel Position
+
+**Summary**
+
+Proposal to introduce a programmatic way to query the Side panel position.
+
+**Document Metadata**
+
+**Author:** [hharshas](https://github.com/hharshas)
+
+**Sponsoring Browser:** Chromium
+
+**Contributors:** [hharshas](https://github.com/hharshas)
+([Summer Of Code](https://summerofcode.withgoogle.com/)), mentor: oliverdunk, solomonkinard
+
+**Created:** 2025-05-27
+
+**Related Issues:** [#chromium](https://issues.chromium.org/issues/406511291)
+
+## Motivation
+
+### Objective
+
+* At `chrome://settings`, it is possible to change the "Side panel position"
+  which affects which side of the browser the side panel UI opens on. This
+  can include panels registered by the developer using the sidePanel API.
+* We should add a way to query this so the developer can adjust their UI
+  based on the position.
+
+#### Use Cases
+
+Programmatic Control: 
+* For example, having phrasing or visuals that depend on the relative position of the page.
+
+### Known Consumers
+
+The Chromium bug has some amount of developer interest. This item is under
+consideration by Chrome among work items for Summer of Code.
+
+## Specification
+
+### Schema
+
+```typescript
+namespace sidePanel {
+ namespace sidePanel {
+  // Possible side panel positions
+  enum PanelPosition {
+    LEFT = "left",
+    RIGHT = "right"
+  };
+
+  interface Functions {
+    // Returns the current position of the side panel.
+    static Promise<PanelPosition> getPosition();
+    
+    // Alternative approach for future extensibility (open for discussion).
+    dictionary PanelSettings {
```

Comment:
It looks like there is a mix of TypeScript and IDL-like syntax here. Can you adjust this to be valid TypeScript, for example removing the Functions interface and using a type or interface for PanelSettings? Feel free to reach out to me or Solomon if you need assistance.